src/base/utils/string.cpp

```diff
@@ -94,7 +94,14 @@ namespace
                     // Both are digits, compare the numbers
                     const auto consumeNumber = [](const QString &str, int &pos) -> int
                     {
-                        const int start = pos;
+                        int start = pos;
+                        // If the partial common prefix ends in digits
+                        // we must include the previous digits here
+                        // in order to convert and sort the number correctly
+                        if (pos > 0)
+                            while ((start >= 0) && str[start].isDigit())
+                                --start;
```

Comment:
Maybe this alone is sufficient (without `if()`) 
```c++
while ((start > 0) && str[start].isDigit())
    --start;
```

Reply:
Damn. This diff is the result of several permutations. I missed this redundancy.
I now see that my code can result in a negative `start`.
Your code can result in off-by-one error.